types/video.js/index.d.ts

```diff
@@ -15,285 +15,7136 @@
+		 *
+		 * @return {Object}
+		 *         A new object of `this.options_` and `obj` merged together.
+		 *
+		 * @deprecated since version 5
+		 */
+		options(obj: any): any;
+
+		played(): TimeRanges;
+
+		/**
+		 * Return the {@link Player} that the `Component` has attached to.
+		 *
+		 * @return {Player}
+		 *         The player that this `Component` has attached to.
+		 */
+		player(): Player;
+
+		/**
+		 * Bind a listener to the component's ready state.
+		 * Different from event listeners in that if the ready event has already happened
+		 * it will trigger the function immediately.
+		 *
+		 * @return {Component}
+		 *         Returns itself; method can be chained.
+		 */
+		ready(callback: (this: Player) => void): this;
+
+		/**
+		 * Remove an attribute from the `Component`s element.
+		 *
+		 * @param {string} attribute
+		 *        Name of the attribute to remove.
+		 *
+		 * @see [DOM API]{@link https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute}
+		 */
+		removeAttribute(attribute: string): void;
+
+		/**
+		 * Remove a child `Component` from this `Component`s list of children. Also removes
+		 * the child `Component`s element from this `Component`s element.
+		 *
+		 * @param {Component} component
+		 *        The child `Component` to remove.
+		 */
+		removeChild(component: Component): void;
+
+		/**
+		 * Remove a CSS class name from the `Component`s element.
+		 *
+		 * @param {string} classToRemove
+		 *        CSS class name to remove
+		 */
+		removeClass(classToRemove: string): void;
+
+		/**
+		 * Queues up a callback to be passed to requestAnimationFrame (rAF), but
+		 * with a few extra bonuses:
+		 *
+		 * - Supports browsers that do not support rAF by falling back to
+		 *   {@link Component#setTimeout}.
+		 *
+		 * - The callback is turned into a {@link Component~GenericCallback} (i.e.
+		 *   bound to the component).
+		 *
+		 * - Automatic cancellation of the rAF callback is handled if the component
+		 *   is disposed before it is called.
+		 *
+		 * @param  {Component~GenericCallback} fn
+		 *         A function that will be bound to this component and executed just
+		 *         before the browser's next repaint.
+		 *
+		 * @return {number}
+		 *         Returns an rAF ID that gets used to identify the timeout. It can
+		 *         also be used in {@link Component#cancelAnimationFrame} to cancel
+		 *         the animation frame callback.
+		 *
+		 * @listens Component#dispose
+		 * @see [Similar to]{@link https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame}
+		 */
+		requestAnimationFrame(fn: Component.GenericCallback): number;
+
+		/**
+		 * Set the value of an attribute on the `Component`'s element
+		 *
+		 * @param {string} attribute
+		 *        Name of the attribute to set.
+		 *
+		 * @param {string} value
+		 *        Value to set the attribute to.
+		 *
+		 * @see [DOM API]{@link https://developer.mozilla.org/en-US
```

Comment:
I don't think this it's a good idea to add type annotations to the JSDoc -- only the type annotations in the TypeScript code are actually checked, so the ones in JSDoc would be misleading.

Reply:
The main purpose was to get access to the documentation faster. I just copied and pasted from the source. What do you suggest?

Reply:
There should be tools, that can migrate types from JSDoc to TS, VSCode can do it. I am not sure but `tsc --declaration` might do it too

Reply:
What --declaration option generates the d.ts files from typescript code. That's not the case here. Anyway there's [dts-gen](https://github.com/Microsoft/dts-gen) which generates d.ts files from js code. I already executed it against video.js but did not helped at all

Reply:
I would remove the copied documentation that's redundant in a typed language -- any `{Type}` should be moved to a `: Type`.

Reply:
so, for instance, this:
``` 
* @param {string|Element} id
*        Video element or video element ID
```
would become this:
``` 
* @param id
*        Video element or video element ID
```
right?

Reply:
Right. The `no-redundant-jsdoc-2` lint rule should catch things like this too.